Libraries/LibWeb/XML/XMLDocumentBuilder.cpp

```diff
@@ -338,6 +357,24 @@ void XMLDocumentBuilder::document_end()
     // The Document is now ready for post-load tasks.
     m_document->set_ready_for_post_load_tasks(true);
 }
+
+Optional<FlyString> XMLDocumentBuilder::namespace_for_element(XML::Name const& element_name)
+{
+    Optional<ByteString> prefix;
+    if (auto parts = element_name.split_limit(':', 2); parts.size() == 2) {
+        prefix = parts[0];
+    }
+
+    auto stack_entry = m_namespace_stack.last();
```

Comment:
This only evaluates the last declaration of namespaces in the stack, what happens if a prefix from higher up in the stack is used?

Reply:
Excellent point - my latest push walks the stack rather than just looking at the last entry.